colossalai/inference/hybridengine/engine.py

```diff
@@ -71,12 +75,19 @@ def __init__(
         do_sample: bool = False,
         num_beams: int = 1,
     ) -> None:
-        assert pp_model or (model and model_policy), "Either pp_model or model with model_policy should be provided."
+        assert model.__class__.__name__ in _supported_models, f"Model {model.__class__.__name__} is not supported."
+        assert (
+            tp_size * pp_size == dist.get_world_size()
+        ), f"TP size({tp_size}) * PP size({pp_size}) should be equal to the global world size ({dist.get_world_size()})"
+        assert model and model_policy, "Model with model_policy should be provided."
         assert dtype in ["fp16", "fp32", "bf16"], "dtype should be one of 'fp16', 'fp32', 'bf16'"
 
-        max_output_len = max(max_output_len, max_input_len + new_length)
+        assert max_batch_size <= 64, "Max batch size exceeds the constraint"
+        assert max_input_len + max_output_len <= 4096, "Max length exceeds the constraint"
 
         self.pp_size = pp_size
+        self.tp_size = tp_size
+
```

Comment:
can pipeline and tp-engine use together? If not, probably added a checking function to check: if pp_size > 1, assert tp_size == 1, and if tp_size > 1, assert pp_size != 1

Reply:
ok